<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>RemoteLinux::GenericLinuxDeviceConfigurationWidget</class>
 <widget class="QWidget" name="RemoteLinux::GenericLinuxDeviceConfigurationWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>556</width>
    <height>309</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string/>
  </property>
  <layout class="QFormLayout" name="formLayout">
   <property name="fieldGrowthPolicy">
    <enum>QFormLayout::FieldsStayAtSizeHint</enum>
   </property>
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item row="1" column="0">
    <widget class="QLabel" name="authTypeLabel">
     <property name="text">
      <string>Authentication type:</string>
     </property>
    </widget>
   </item>
   <item row="1" column="1">
    <widget class="QWidget" name="authTypeButtonsWidget" native="true">
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QRadioButton" name="passwordButton">
        <property name="text">
         <string>Password</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="keyButton">
        <property name="text">
         <string>&amp;Key</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="agentButton">
        <property name="text">
         <string>Key via ssh-agent</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_4">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QLabel" name="hostNameLabel">
     <property name="text">
      <string>&amp;Host name:</string>
     </property>
     <property name="buddy">
      <cstring>hostLineEdit</cstring>
     </property>
    </widget>
   </item>
   <item row="2" column="1">
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <widget class="QLineEdit" name="hostLineEdit">
       <property name="placeholderText">
        <string>IP or host name of the device</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="sshPortLabel">
       <property name="text">
        <string>&amp;SSH port:</string>
       </property>
       <property name="buddy">
        <cstring>sshPortSpinBox</cstring>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="sshPortSpinBox">
       <property name="minimum">
        <number>0</number>
       </property>
       <property name="maximum">
        <number>65535</number>
       </property>
       <property name="value">
        <number>22</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="hostKeyCheckBox">
       <property name="text">
        <string>&amp;Check host key</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="3" column="0">
    <widget class="QLabel" name="freePortsLabel">
     <property name="text">
      <string>Free ports:</string>
     </property>
     <property name="buddy">
      <cstring>portsLineEdit</cstring>
     </property>
    </widget>
   </item>
   <item row="3" column="1">
    <layout class="QHBoxLayout" name="horizontalLayout_7">
     <item>
      <widget class="QLineEdit" name="portsLineEdit">
       <property name="toolTip">
        <string>You can enter lists and ranges like this: '1024,1026-1028,1030'.</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="portsWarningLabel">
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="connectionTimeoutLabel">
       <property name="text">
        <string>Timeout:</string>
       </property>
       <property name="buddy">
        <cstring>timeoutSpinBox</cstring>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="timeoutSpinBox">
       <property name="suffix">
        <string>s</string>
       </property>
       <property name="maximum">
        <number>10000</number>
       </property>
       <property name="singleStep">
        <number>10</number>
       </property>
       <property name="value">
        <number>1000</number>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="4" column="0">
    <widget class="QLabel" name="userNameLabel">
     <property name="text">
      <string>&amp;Username:</string>
     </property>
     <property name="buddy">
      <cstring>userLineEdit</cstring>
     </property>
    </widget>
   </item>
   <item row="4" column="1">
    <widget class="QLineEdit" name="userLineEdit"/>
   </item>
   <item row="5" column="0">
    <widget class="QLabel" name="passwordLabel">
     <property name="text">
      <string>&amp;Password:</string>
     </property>
     <property name="buddy">
      <cstring>pwdLineEdit</cstring>
     </property>
    </widget>
   </item>
   <item row="5" column="1">
    <layout class="QHBoxLayout" name="horizontalLayout_6">
     <item>
      <widget class="QLineEdit" name="pwdLineEdit">
       <property name="echoMode">
        <enum>QLineEdit::Password</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="showPasswordCheckBox">
       <property name="text">
        <string>Show password</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="6" column="0">
    <widget class="QLabel" name="keyLabel">
     <property name="text">
      <string>Private key file:</string>
     </property>
    </widget>
   </item>
   <item row="6" column="1">
    <layout class="QHBoxLayout" name="horizontalLayout_8">
     <item>
      <widget class="Utils::PathChooser" name="keyFileLineEdit" native="true"/>
     </item>
     <item>
      <widget class="QPushButton" name="createKeyButton">
       <property name="text">
        <string>Create New...</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="0" column="0">
    <widget class="QLabel" name="machineTypeLabel">
     <property name="text">
      <string>Machine type:</string>
     </property>
    </widget>
   </item>
   <item row="0" column="1">
    <widget class="QLabel" name="machineTypeValueLabel"/>
   </item>
   <item row="7" column="0">
    <widget class="QLabel" name="gdbServerLabel">
     <property name="text">
      <string>GDB server executable:</string>
     </property>
    </widget>
   </item>
   <item row="7" column="1">
    <widget class="QLineEdit" name="gdbServerLineEdit">
     <property name="placeholderText">
      <string>Leave empty to look up executable in $PATH</string>
     </property>
    </widget>
   </item>
  </layout>
  <zorder>passwordLabel</zorder>
  <zorder>authTypeButtonsWidget</zorder>
  <zorder>hostNameLabel</zorder>
  <zorder>userNameLabel</zorder>
  <zorder>freePortsLabel</zorder>
  <zorder>userLineEdit</zorder>
  <zorder>authTypeLabel</zorder>
  <zorder>keyLabel</zorder>
  <zorder>machineTypeLabel</zorder>
  <zorder>machineTypeValueLabel</zorder>
  <zorder>gdbServerLabel</zorder>
  <zorder>gdbServerLineEdit</zorder>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Utils::PathChooser</class>
   <extends>QWidget</extends>
   <header location="global">utils/pathchooser.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>passwordButton</tabstop>
  <tabstop>keyButton</tabstop>
  <tabstop>hostLineEdit</tabstop>
  <tabstop>sshPortSpinBox</tabstop>
  <tabstop>hostKeyCheckBox</tabstop>
  <tabstop>portsLineEdit</tabstop>
  <tabstop>timeoutSpinBox</tabstop>
  <tabstop>userLineEdit</tabstop>
  <tabstop>pwdLineEdit</tabstop>
  <tabstop>showPasswordCheckBox</tabstop>
  <tabstop>createKeyButton</tabstop>
  <tabstop>gdbServerLineEdit</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
